app/main.py

```diff
@@ -1,8 +1,19 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result = [Person(person["name"], person["age"]) for person in people]
+    for i, person in enumerate(people):
+        if "wife" in person and person["wife"] is not None:
+            Person.people[person["name"]].wife = Person.people[person["wife"]]
```

Comment:
You could make code more efficient using walrus operator and get. Besides that, you could retrieve person instance from the beginning and later refer to it
```suggestion
        person_instance = Person.people[person["name"]]
        if wife_name := person.get("wife"):
            person_instance.wife = Person.people[wife_name]
```
Apply the similar change for husband condition as well